clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/PutIfAbsentOperation.java

```diff
@@ -32,10 +40,11 @@ public OperationCode getOpCode() {
    * for the same key.
    */
   @Override
-  public Operation<K> apply(final Operation<K> previousOperation) {
+  public Operation<K, V> apply(final Operation<K, V> previousOperation) {
     if(previousOperation == null) {
       return this;
     } else {
+      assertSameKey(previousOperation);
       return previousOperation;
```

Comment:
That's exactly what lines 44-45 takes care of. If the application of _previousOperation_ results in a null, for example in a sequence like PUT, REMOVE, PUT_IF_ABSENT, the resolution goes as follow: 
PUT applied on top of null(initial state) -> PUT
REMOVE applied on top of the the previous result(PUT in this case) -> null
PUT_IF_ABSENT applied on top of null(the previous result) -> PUT_IF_ABSENT

So finally the PUT_IF_ABSENT succeeded. And that's exactly what the expected result is.